mono/mini/gshared.cs

```diff
@@ -1058,6 +1058,14 @@ public enum AnEnum {
 		return 0;
 	}
 
+	public static int test_0_contains_key_constrained_get_hash () {
+		IDictionary<Type, EmptyStruct> dict = new Dictionary<Type, EmptyStruct> ();
```

Comment:
Would be nice to have a lower level test.

Reply:
Made one:
https://gist.github.com/vargaz/a55dbc9cafefd6e8a34d7575872363c1

Reply:
Works, switched test.